### PR TITLE
CI - trading repeatability for speed

### PR DESCRIPTION
**Pull request by pshomov**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -433,7 +424,15 @@ jobs:
   failure-notification:
     runs-on: ubuntu-latest
     if: failure()
-    needs: [docker-express, docker-next, prepare, linting, formatting, trigger-deployments]
+    needs:
+      [
```
Comment (sindrig) [inline .github/workflows/build.yml:428]: Consider using native yaml list:
```
needs:
  - docker-express
  - docker-next
  ...
```

File: scripts/check_formatting.sh [hunk 2]
```diff
@@ -5,6 +5,10 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
 source $DIR/_common.sh
 
+exec yarn run \
```
Comment (sindrig) [inline scripts/check_formatting.sh:8]: Out of curiosity, why are we using `exec`? Also probably just want to either remove the `exec docker buildx` below or at least comment it out.

File: scripts/lint.sh [hunk 3]
```diff
@@ -5,7 +5,10 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
 source $DIR/_common.sh
 
+exec yarn run \
+  lint ${APP}
 
+# Docker based run, not used at the moment due to severe speed penalty
```
Comment (sindrig) [inline scripts/lint.sh:11]: ditto

File: scripts/test.sh [hunk 4]
```diff
@@ -37,10 +26,25 @@ if [ -f $PROJECT_ROOT/$APP_HOME/docker-compose.ci.yml ]; then
...
+    $PROJECT_ROOT
+
   # Running the tests using docker-compose
   SUT=${DOCKER_REGISTRY}${RUNNER}:${DOCKER_TAG} docker-compose -p test-$APP $COMPOSE_FILES run --rm sut
 else
   # Standalone execution of tests when no external dependencies are needed (DBs, queues, etc.)
+  exec yarn run \
+    test ${APP}
+
+  # Docker based run, not used at the moment due to severe speed penalty
```
Comment (sindrig) [inline scripts/test.sh:47]: ditto

**Review by sindrig — DISMISSED** [review]
:+1:

File: scripts/check_formatting.sh (hunk 2 above)
Comment (pshomov) [reply scripts/check_formatting.sh:8]: done

File: .github/workflows/build.yml (hunk 1 above)
Comment (pshomov) [reply .github/workflows/build.yml]: reverted change, not relevant to the issue at hand

**Review by sindrig — APPROVED** [review]

**Pull request merged**